test/ref/reduce_mean.cpp

```diff
@@ -62,6 +87,31 @@ TEST_CASE(reduce_mean_axis1)
     EXPECT(results_vector == gold);
 }
 
+TEST_CASE(reduce_mean_variable_axis1)
```

Comment:
Is a test on `axis = 1` possible with a multidimensional input, with` keep_dims=1`? For example, this is what I am cutting and pasting from Netron.  And/or -- but near identical -- test -- with `axis = -2` instead of 1. Thanks.

```
shape = [3, 2, 2]
axes = np.array([1], dtype=np.int64)
keepdims = 1

node = onnx.helper.make_node(
    "ReduceMean",
    inputs=["data", "axes"],
    outputs=["reduced"],
    keepdims=keepdims,
)

data = np.array(
    [[[5, 1], [20, 2]], [[30, 1], [40, 2]], [[55, 1], [60, 2]]],
    dtype=np.float32,
)
reduced = np.mean(data, axis=tuple(axes), keepdims=keepdims == 1)
# print(reduced)
# [[[12.5, 1.5]]
# [[35., 1.5]]
# [[57.5, 1.5]]]

expect(
    node,
    inputs=[data, axes],
    outputs=[reduced],
    name="test_reduce_mean_keepdims_example",
)

np.random.seed(0)
data = np.random.uniform(-10, 10, shape).astype(np.float32)
reduced = np.mean(data, axis=tuple(axes), keepdims=keepdims == 1)

expect(
    node,
    inputs=[data, axes],
    outputs=[reduced],
    name="test_reduce_mean_keepdims_random",
)
```

Reply:
I'm not quite catching your drift here. As for keepdims, that cannot be tested in the reference tests, as it is not a part of the operator itself, but is accomplished by inserting a squeeze operator after the reduction.

Reply:
I am not sure, in a black box testing it matters if there one or instead two moving parts inside.. It is the results that matter, and so if the squeeze operator is getting inserted, that is fine by me.. 

Maybe I am missing something here? Thanks.

Reply:
I see what you mean. That's for the onnx/verify tests to check. I had already added a couple for the case when axes are a variable dynamic input, but I hadn't for any of the others. An oversight on my part, I'll add a couple more onnx/verify tests.